### PR TITLE
Add package MathGL (version 2.3.5.1)

### PR DESCRIPTION
**Pull request by dsogari**
MathGL from http://mathgl.sourceforge.net/doc_en/Main.html

There are many optional dependencies that I made requirements (for the sake of simplicity, since I do not know exactly which of them are truly optional and which are needed at run-time).

### CODE REVIEW

File: mingw-w64-mathgl/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+_realname=mathgl
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=2.3.5.1
+pkgrel=1
+pkgdesc="Library for high-quality scientific graphics (mingw-w64)"
+arch=('any')
+url="http://mathgl.sourceforge.net/"
+license=('LGPL')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
```
Comment (mati865) [inline mingw-w64-mathgl/PKGBUILD]: `"${MINGW_PACKAGE_PREFIX}-gcc"` is part of `${MINGW_PACKAGE_PREFIX}-toolchain` therefore it's redundant.

File: mingw-w64-mathgl/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+         "${MINGW_PACKAGE_PREFIX}-fltk"
+         "${MINGW_PACKAGE_PREFIX}-libharu"
+         "${MINGW_PACKAGE_PREFIX}-libjpeg-turbo"
+         "${MINGW_PACKAGE_PREFIX}-libpng"
+         "${MINGW_PACKAGE_PREFIX}-giflib"
+         "${MINGW_PACKAGE_PREFIX}-qt5"
+         "${MINGW_PACKAGE_PREFIX}-freeglut"
+         "${MINGW_PACKAGE_PREFIX}-wxWidgets")
+source=("${_realname}-${pkgver}.tar.gz::http://downloads.sourceforge.net/${_realname}/${_realname}-${pkgver}.tar.gz"
+        "${_realname}-${pkgver}.patch")
```
Comment (mati865) [inline mingw-w64-mathgl/PKGBUILD]: Use static name for patch.

File: mingw-w64-mathgl/PKGBUILD (hunk 1 above)
Comment (dsogari) [inline mingw-w64-mathgl/PKGBUILD]: Ok, thanks! Will update it

**Comment by dsogari** [comment]
@mati865 
Do you think I should create different packages for GPL and LGPL versions of the library?

I need the LPGL version, but there may be users who want the GPL version, which has support for GSL (https://www.gnu.org/software/gsl/).

**Comment by mati865** [comment]
@dsogari I haven't heard about MathGL before.
Pinging @Alexpux and @mingwandroid

**Pull request merged**